generator/feature_generator.hpp

```diff
@@ -40,6 +40,8 @@ class FeaturesCollector
 
   string const & GetFilePath() const { return m_datFile.GetName(); }
 
+  uint32_t GetNextFeatureId() const { return m_featureID; }
```

Comment:
I cannot. The problem is in uint32_t FeaturesCollector2::operator() (FeatureBuilder2 & fb).
There's "if (fb.PreSerialize(holder.m_buffer))" in it. If fb.PreSerialize(holder.m_buffer) returns false I don't see any way to get feature id except for using GetNextFeatureId().

1. I modified operator() to return feature id;
2. I made GetNextFeatureId() protected;